src/components/tabs/tab-bar.tsx

```diff
@@ -230,11 +228,18 @@ const TabBar = (props: TabBarProps) => {
                 onMouseDown={e => dragMouseDown(e, child)}
                 onMouseUp={closeDragElement}
               >
-                {child.tabComponent.props.text}
+                {child.tabComponent.props.tabHeader || child.tabComponent.props.text}
```

Comment:
this way you're assuming that the `tabHeader` was constructed before the whole tab thing.

what do you think about receiving a "rendering function" so `tabHeader` only gets constructed here along with everything else? what would change:

```jsx
{child.tabComponent.props.tabHeader ? child.tabComponent.props.tabHeader() ? child.tabComponent.props.text}
```
or better, take a look of how react-bootstrap does this similar feature with their `as` prop (so a `Row` can assume other components instead of just `div`s): https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Row.js